protocol/communities_messenger_signers_test.go

```diff
@@ -0,0 +1,229 @@
+package protocol
+
+import (
+	"crypto/ecdsa"
+	"testing"
+
+	gethbridge "github.com/status-im/status-go/eth-node/bridge/geth"
+	"github.com/status-im/status-go/eth-node/crypto"
+	"github.com/status-im/status-go/eth-node/types"
+	"github.com/status-im/status-go/protocol/common"
+	"github.com/status-im/status-go/protocol/communities"
+	"github.com/status-im/status-go/protocol/protobuf"
+	"github.com/status-im/status-go/protocol/requests"
+	"github.com/status-im/status-go/protocol/tt"
+	"github.com/status-im/status-go/waku"
+	"github.com/stretchr/testify/suite"
+	"go.uber.org/zap"
+)
+
+func TestMessengerCommunitiesSignersSuite(t *testing.T) {
+	suite.Run(t, new(MessengerCommunitiesSignersSuite))
+}
+
+type MessengerCommunitiesSignersSuite struct {
+	suite.Suite
+	john  *Messenger
+	bob   *Messenger
+	alice *Messenger
+
+	shh    types.Waku
+	logger *zap.Logger
+}
+
+func (s *MessengerCommunitiesSignersSuite) SetupTest() {
+	s.logger = tt.MustCreateTestLogger()
+
+	config := waku.DefaultConfig
+	config.MinimumAcceptedPoW = 0
+	shh := waku.New(&config, s.logger)
+	s.shh = gethbridge.NewGethWakuWrapper(shh)
+	s.Require().NoError(shh.Start())
+
+	s.john = s.newMessenger()
+	s.bob = s.newMessenger()
+	s.alice = s.newMessenger()
+	_, err := s.john.Start()
+	s.Require().NoError(err)
+	_, err = s.bob.Start()
+	s.Require().NoError(err)
+	_, err = s.alice.Start()
+	s.Require().NoError(err)
+}
+
+func (s *MessengerCommunitiesSignersSuite) TearDownTest() {
+	s.Require().NoError(s.john.Shutdown())
+	s.Require().NoError(s.bob.Shutdown())
+	s.Require().NoError(s.alice.Shutdown())
+	_ = s.logger.Sync()
+}
+
+func (s *MessengerCommunitiesSignersSuite) newMessengerWithKey(privateKey *ecdsa.PrivateKey) *Messenger {
+	messenger, err := newCommunitiesTestMessenger(s.shh, privateKey, s.logger, nil, nil)
+	s.Require().NoError(err)
+
+	return messenger
+}
+
+func (s *MessengerCommunitiesSignersSuite) newMessenger() *Messenger {
+	privateKey, err := crypto.GenerateKey()
+	s.Require().NoError(err)
+
+	return s.newMessengerWithKey(privateKey)
+}
+
+func (s *MessengerCommunitiesSignersSuite) createCommunity(controlNode *Messenger) *communities.Community {
+	community, _ := createCommunity(&s.Suite, controlNode)
+	return community
+}
+
+func (s *MessengerCommunitiesSignersSuite) advertiseCommunityTo(controlNode *Messenger, community *communities.Community, user *Messenger) {
+	advertiseCommunityTo(&s.Suite, community, controlNode, user)
+}
+
+func (s *MessengerCommunitiesSignersSuite) joinCommunity(controlNode *Messenger, community *communities.Community, user *Messenger) {
+	request := &requests.RequestToJoinCommunity{CommunityID: community.ID()}
+	joinCommunity(&s.Suite, community, controlNode, user, request)
+}
+
+func (s *MessengerCommunitiesSignersSuite) TestControlNodeUpdate() {
```

Comment:
This is the proof of concept. Scenario:

// john as control node publishes community update
// bob accepts community update
// alice accepts community update
// make bob the control node
// make alice aware of control node change
// john can still publish community update, he doesn't know about control node change yet
// bob rejects community update as it is not signed by control node
// alice rejects community update as it is not signed by control node
// make john aware of control node change
// john can't publish community update anymore
// bob as control node publishes community update
// john accepts community update
// alice accepts community update